clarify event-time requirements (#7148)
dbt-labs/docs.getdbt.com: website/docs/docs/build/incremental-microbatch.md

Incremental models in dbt are a [materialization](/docs/build/materializations) designed to efficiently update your data warehouse tables by only transforming and loading _new or changed data_ since the last run. Instead of reprocessing an entire dataset every time, incremental models process a smaller number of rows, and then append, update, or replace those rows in the existing table. This can significantly reduce the time and resources required for your data transformations.

Microbatch is an incremental strategy designed for large time-series datasets:
- It relies solely on a time column ([`event_time`](/reference/resource-configs/event-time)) to define time-based ranges for filtering. 
- Set the `event_time` column for your microbatch model and its direct parents (upstream models). Note, this is different to `partition_by`, which groups rows into partitions.
  :::caution Required

>   If your upstream models don't have `event_time` configured, dbt _cannot_ automatically filter them during batch processing and will perform full table scans on every batch run. To avoid this, configure `event_time` on every upstream model that should be filtered. If you want to exclude a model from auto-filtering, see [Opt out of auto-filtering](#opting-out-of-auto-filtering) for how to opt out.

Reviewer: ```suggestion
  If your upstream models don't have `event_time` configured, dbt _cannot_ automatically filter them during batch processing and will perform full table scans on every batch run. To avoid this, configure `event_time` on every upstream model that should be filtered. Learn how to exclude a model from auto-filtering by [opting out of auto-filtering](#opting-out-of-auto-filtering).
```

>   If your upstream models don't have `event_time` configured, dbt _cannot_ automatically filter them during batch processing and will perform full table scans on every batch run. To avoid this, configure `event_time` on every upstream model that should be filtered. If you want to exclude a model from auto-filtering, see [Opt out of auto-filtering](#opting-out-of-auto-filtering) for how to opt out.

Reviewer: Later in this page we say this, so I wonder if we should also link to other incremental strategies? If they don't have event_time configured in their upstream models, maybe microbatch is not for them?

>Note, microbatch might not be the best strategy for all use cases. Consider other strategies for use cases such as not having a reliable event_time column or if you want more control over the incremental logic. Read more in [How microbatch compares to other incremental strategies](https://docs-getdbt-com-git-clarify-event-time-dbt-labs.vercel.app/docs/build/incremental-microbatch#how-microbatch-compares-to-other-incremental-strategies).
